core/src/main/java/nl/nn/adapterframework/webcontrol/api/ServletDispatcher.java

```diff
@@ -121,23 +118,25 @@ public void service(HttpServletRequest request, HttpServletResponse response) th
 		}
 		else {
 			//Always add the cors headers when the origin has been set
-			if(allowedCorsDomains.contains(origin)) {
+			allowedCorsDomains.stream().filter(s -> allowedCorsDomains.contains(origin)).findAny().map(s -> {
```

Comment:
You don't have to loop through the allowedCorsDomains list. It's just a safety check to determine whether or not to add the preflight CORS headers.